20250429-RS-scheduler-broadcast-job-type.md

```diff
@@ -0,0 +1,320 @@
+  // client.
+  BroadcastJobEventType type = 1;
+
+  // data contains the actual data of the job. It may be wrapped in another
+  // message, depending on the type of broadcast job.
+  google.protobuf.Any data = 2;
+}
+
+// TargetBroadcast are job types which are "broadcast", meaning they are sent
+// to all clients who support this target on connection, and when the job is
+// created. On delete, the job is signalled to all matching clients that the
+// job has been deleted with an empty data payload.
+message TargetBroadcast {
+  oneof broadcast {
+    // TargetActorSubscription is the message which holds broadcast component spec
+    // for an Actor Subscription.
+    TargetActorSubscription actor_subscription = 1;
+
+    // TargetDurableActorID is the message which holds broadcast durable actor
+    // ID for an Actor Subscription.
+    // The data field of the job must be of type `common.InvokeRequest`, where
+    // the `.HttpExtension.Verb` will always be ignored, in favour of a POST
+    // when the durable actor is invoked.
+    TargetDurableActorID durable_actor_id = 2;
+  }
+}
+
+// TargetActorSubscription is the message which holds broadcast component spec
+// for an Actor Subscription.
+message TargetActorSubscription {
+  // id is the actor ID.
+  string id = 1;
+
+  // type is the actor type.
+  string type = 2;
+}
+
+// TargetDurableActorID is the message which holds broadcast durable actor ID
+// jobs.
+message TargetDurableActorID {
+  // id is the actor ID.
+  string id = 1;
+
+  // type is the actor type.
+  string type = 2;
+}
+```
+
+## Use case examples
+
+### Actor PubSub Subscription
+
+As detailed in the [Proposal](https://github.com/dapr/proposals/pull/78), the Actor PubSub subscription object which is manifest which details how a subscription should be defined for receiving messages to a single actor ID.
+In order for this manifest to be durable, it needs to be broadcast to all clients which implement that Actor type.
+Upon receiving the object, or during dissemination, each actor runtime can determine which set of subscription manifests are assigned to that host based on an Actor ID placement table lookup.
+The object subscription object itself should have no definitive lifetime, and is only created and deleted by the application.
+It could be the case that these objects also have a corresponding TTL associated.
+
+The actor runtime will receive events about the job;
+- on put events the actor runtime will create the underlying subscription, or replace one that is already existing.
+- on delete events the actor runtime will delete the subscription.
+- there should not be any trigger events.
+
+The Actor Subscription broadcast job will take the following form.
+Notice the schedule is set to infinite as the Job should never be triggered.
+The name of the Job object can be any application defined name.
+This job type will return an `already exists` error if a PUT is attempted on an existing Actor Subscription job.
```

Comment:
Good - this aligns with my earlier comment (which should probably be changed then to remove "update").